src/atom-config/make-handler/lint-html.js

```diff
@@ -0,0 +1,24 @@
+'use babel'
+
+import { spliceReplace1 } from '../../m-utils'
+
+
+const hasHtmlScope = config =>
+  config.scopes.indexOf(config.embeddedScope) !== -1
+
+/* eslint-disable no-param-reassign */
+
+const setLintHtml = config => (value) => {
+  const { scopes, embeddedScope } = config
+  // set lintHtmlFiles
+  config.lintHtmlFiles = value
+  // add html to scopes if missing
```

Comment:
I used the blanket disable for improved readability. It's isolated to the file, so has limited risk of leading to harmful code. If you are really _that_ against it,  I can use next-line. But this file particularly highlights the value of using blanket disable _while in a limited scope_. To use next-line disable here means I need to put it in between a _real_ comment and the line that comment describes.

Had this been a larger file, my inclination here would be to put a disable above the function and an enable after the function, but since at the end of the function, we exit the file, there's no reason to enable it again.

---
I feel very uncomfortable with functions not returning anything. If I don't have an obvious return value, I try to return something that seems potentially useful. It increases the likelihood that I can change the context of how I use the function in the future and have it "just work" without having to dig into the function's internals. The config returned is _not_ a "new" config. It is the same config we recieved at the top.

Reply:
"add" -> "Add" for sure. Also "set" -"Set".

Not a huge fan of all-caps 'HTML'. I think any remaining need to _always_ capitalize that was invalidated when html5 standards were released with instructions to set your doctype with `<!doctype html>`. Of note here is that we are not really referencing the html language here, so much as the scope `source.js.embedded.html` which has no capital letters in it.

Reply:
Lint rules are there for a reason, so any time they are disabled should be treated with caution, or the lint configuration revisited. Blanket disables are particularly dangerous since you then have nothing marking _why_ it has been disabled, or where this disabling was needed. This can lead to issues with the linter not catching things that are problems elsewhere from where you intended.

For this particular case? Sure it makes no _real_ difference, but it's a bad habit that shouldn't be encouraged.

----

> It increases the likelihood that I can change the context of how I use the function in the future and have it "just work" without having to dig into the function's internals. The config returned is not a "new" config. It is the same config we recieved at the top.

That makes some sense, it's just odd to see a function that is purely there for the side-effects to return a value.

Reply:
> Not a huge fan of all-caps 'HTML'. I think any remaining need to _always_ capitalize that was invalidated when html5 standards were released with instructions to set your doctype with `<!doctype html>`.

We aren't talking about code here, but the _name_ of the standard, which is still (abbreviated to) `HTML` 😉.

Reply:
> We aren't talking about code here, but the name of the standard
Really? We aren't? When _I wrote the comment_, I'm pretty sure I was short-handing for `source.js.embedded.html`. If that was unclear, then I guess it should say
```
// Add source.js.embedded.html to scopes if missing
```

Reply:
> For this particular case? Sure it makes no real difference, but it's a bad habit that shouldn't be encouraged.

I'm not sold. But I'll concede. Not one I care about enough to argue about it any further than to explain my choice, which I've done already.

> That makes some sense, it's just odd to see a function that is purely there for the side-effects to return a value.

I can understand that, but you have to remember I'm coming from a place where it's odd to write a function purely for side-effects. This sort-of helps bridge the gap a little.

Knowing there is a return value leaves me with a bit of extra hope that maybe in the future I'll be able to use that return value as a stepping-stone to factoring away the side-effect. 🌈 🦄

Reply:
That sounds like a good change, I thought you were referring to "HTML" and using that to imply the full scope name for the embedded JS part. 😆